pkg/plugins/platform/local.go

```diff
@@ -980,6 +1002,18 @@ func GetActiveApp(siteName string) (App, error) {
 		return app, err
 	}
 
-	err = app.Init(activeAppRoot)
-	return app, err
+	_ = app.Init(activeAppRoot)
+	// Make sure AppConfig.Name is set in case this app is being used for Cleanup().
+	if app.GetName() == "" {
+		app, _ := app.(*LocalApp)
+		app.AppConfig.Name = siteName
+		// Since the site may be used for Cleanup(), also check that AppConfig.DataDir is set.
+		// This ensures that the sites data can be removed with the containers.
+		if app.AppConfig.DataDir == "" {
+			dataDir := fmt.Sprintf("%s/%s", util.GetGlobalDdevDir(), app.AppConfig.Name)
```

Comment:
Is siteName aka AppConfig.Name guaranteed to be populated and not empty? If it were somehow empty seems like we might delete ~/.ddev ?

Reply:
I could see it potentially being empty if someone were to run ddev rm by itself with no site-name argument in a directory with a deleted .ddev/config.yml. Not quite sure how that would play out.